### ISSUE TITLE
[Stage 2] Training collapse with NaN loss

### ISSUE THREAD

**Issue by Y-Sisyphus** [bytetriper/RAE#26]
Awesome work, thanks for releasing the project and models.
I have similar questions as [issue16](https://github.com/bytetriper/RAE/issues/16.) I adopt the [default config](https://github.com/bytetriper/RAE/blob/main/configs/stage2/training/ImageNet256/DiTDH-XL_DINOv2-B.yaml) in the repo by using 32 GPUs. NaN appears at iteration 67k. Here is the log.

<img width="907" height="480" alt="Image" src="https://github.com/user-attachments/assets/98af92dd-6f90-41fb-94e4-de78d06b23ce" />

Do you have any suggestions? Thank you in advance for your time and assistance!

**Comment by Y-Sisyphus** [comment]
> hello, I encounter the same error and change the precision from bp16 to fp32 resolves the problem

Thank you! I will try.

**Comment by bytetriper** [comment]
We're still debugging -- but it's reported than changing PyTorch version to 2.8.0 (CUDA 12.4) help resolving the issue on H100s.

**Comment by ZHANG-GuiGui** [comment]
Hey,

Thanks for releasing the code and very impressive work! I had some difficulties in reproducing the results in Table 8 Generation@256 w/o guidance.

I was using fp32 and no gradient accumulation, total_bsz=1024, trained for 80 epochs along with the provided autoencoder ckpt. Following the evaluation process https://github.com/bytetriper/RAE/issues/11, here is what i got:

```
Inception Score: 215.77584838867188
FID: 3.1104738435340664
sFID: 8.010660241227924
Precision: 0.75632
Recall: 0.5847716954339087
```

Do you have any idea where it could go wrong?

Thank you!

**Comment by sen-ye** [comment]
enabling qk norm in dit also helps stablize the training

**Comment by bytetriper** [comment]
We've updated the codebase and done sanity checks!

On the updated codebase, we ran a DiTDH-XL (fp32) on DINOv2-B for ~200 epochs without seeing any loss spikes.
